### PR TITLE
Activity tab in files sidebar

### PR DESCRIPTION
**Pull request by PVince81**
@nickvergessen FYI

Not sure what attributes the endpoint is going to return eventually.
Everything is a dummy at the moment.

- [x] REQUIRES https://github.com/owncloud/core/pull/18178
- [x] REQUIRES https://github.com/owncloud/core/pull/18290

### CODE REVIEW

**Comment by PVince81** [comment]
For https://github.com/owncloud/core/issues/18164

**Comment by karlitschek** [comment]
You guys rock!!  :-)

**Comment by PVince81** [comment]
also requires https://github.com/owncloud/core/pull/18290

**Comment by PVince81** [comment]
@nickvergessen I fixed the spinner issue.
Also I made ActivityCollection more generic to accept "objecttype" and "objectid".
Please implement the backend to make the filter work :smile: 

Currently it looks like this using the existing CSS from the page:
activity-nofilter

**Comment by jancborchardt** [comment]
Nice! :) Of course the activities shown in there should only be the ones pertaining to that specific file. :)
In the screenshot above: Not the image addition or the folder creation – we need to keep it clean.

File: appinfo/app.php [hunk 1]
```diff
@@ -49,3 +49,12 @@
 // Cron job for sending emails and pruning the activity list
 $c->getServer()->getJobList()->add('OCA\Activity\BackgroundJob\EmailNotification');
 $c->getServer()->getJobList()->add('OCA\Activity\BackgroundJob\ExpireActivities');
+
+// Needed for the files sidebar entry
+\OCP\Util::connectHook(
```
Comment (nickvergessen) [inline appinfo/app.php]: Move to `FilesHooksStatic::register()`

**Comment by PVince81** [comment]
@jancborchardt see previous comment

>Please implement the backend to make the filter work

**Comment by PVince81** [comment]
- [ ] TODO: implement backend for the filter
- [ ] TODO: fix hook

File: js/activitycollection.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+		 * @param {int} objectType file id or null
+		 */
+		setObjectType: function(objectType) {
+			this._objectType = objectType;
+		},
+
+		url: function() {
+			var query = {
+				page: 1,
+				filter: 'all'
```
Comment (nickvergessen) [inline js/activitycollection.js]: filter: 'filter'

Comment (PVince81) [inline js/activitycollection.js]: but only when objectid or objecttype are set ? or always ?

Comment (nickvergessen) [inline js/activitycollection.js]: all gives all, filter uses object id and type, see the following lines:
https://github.com/owncloud/activity/pull/363/files#diff-7842161f69cff3997d7b10126d8c368fR223

**Comment by nickvergessen** [comment]
activity filter option is merged, event listener for loading js/css is merged

if you rebase we can test it all together

**Comment by PVince81** [comment]
Will do

**Comment by PVince81** [comment]
Rebased, fixed the hook and adapted.
It works now.

- [ ] TODO: JS unit tests

**Comment by PVince81** [comment]
Oops, I missed some files

**Comment by PVince81** [comment]
Should be good now.

**Comment by PVince81** [comment]
- [ ] REQUIRES: https://github.com/owncloud/core/pull/18459 for the unit test loading

**Comment by PVince81** [comment]
Unit tests added. But they will only run once https://github.com/owncloud/core/pull/18459 is merged in core.

**Comment by PVince81** [comment]
See how difficult it is to load core's JS files in the right order: https://github.com/owncloud/activity/pull/358/files#diff-6e207d934280c969b0add642e133ee90R29

Fortunately we added "core/js/core.json" that lists the core JS files for that which can be reused here.

File: js/activitycollection.js (hunk 2 above)
Comment (nickvergessen) [inline js/activitycollection.js]: this still needs updating?

Comment (PVince81) [inline js/activitycollection.js]: scroll down

Comment (PVince81) [inline js/activitycollection.js]: we only set the filter if either objectId or objectType are set. Else we fetch all (could be used later in other views)

**Comment by PVince81** [comment]
This is ready for review @jancborchardt @nickvergessen

**Comment by nickvergessen** [comment]
Works :+1:

**Comment by jancborchardt** [comment]
Yup, nice! :)

**Pull request merged**